zxlive/dodo.py

```diff
@@ -0,0 +1,261 @@
+# DODO-GPT is brought to you by Dave the Dodo of Picturing Quantum Processes fame
+
+import pyzx as zx
+import openai
+import sounddevice as sd
+import os
+import numpy as np
+import base64
+import requests
+from pydub import AudioSegment
+from .utils import GET_MODULE_PATH
+from enum import IntEnum
+
+API_KEY = 'no-key'
+
+# Need to add error handling still! (e.g. if invalid key, or if no connection to chat-gpt server, etc.)
+# Maybe add a config (i.e. to set the api_key, the gpt model, the OpenAI Whisper voice, etc.)
+# Maybe also record your full transripts with DODO-GPT (and have the option of whether to save it to file when you close zxlive)
+
+#TEMP... (# TEMP - this should just be calling xz.utils.VertexType instead (not sure why that doesn't work though?): VertexType(1).name)
+class VertexType(IntEnum):
+    """Type of a vertex in the graph."""
+    BOUNDARY = 0
+    Z = 1
+    X = 2
+    H_BOX = 3
+    W_INPUT = 4
+    W_OUTPUT = 5
+    Z_BOX = 6
+    
+#TEMP... (# TEMP - this should just be calling xz.utils.VertexType instead (not sure why that doesn't work though?): VertexType(1).name)
+class EdgeType(IntEnum):
+    """Type of an edge in the graph."""
+    SIMPLE = 1
+    HADAMARD = 2
+    W_IO = 3
+
+def get_local_api_key():
+    """Get the API key from key.txt file"""
+    f = open(GET_MODULE_PATH()+"/user/key.txt", "r")
```

Comment:
It might be better to have the API key be settable in the settings of ZXLive.

Reply:
That was my plan! This is just a quick temporary solution so that I can push the changes. I guess we could also save API keys locally if we hash them first, so that way you won't have to re-enter it in every new instance of the program?